resilience4j-spring/src/main/java/io/github/resilience4j/bulkhead/configure/PlainObjectBulkheadAspectExt.java

```diff
@@ -0,0 +1,139 @@
+import io.github.resilience4j.bulkhead.internal.ThreadPoolBulkheadAdapter;
+import io.github.resilience4j.timelimiter.TimeLimiter;
+import io.github.resilience4j.timelimiter.TimeLimiterRegistry;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.*;
+import java.util.function.Supplier;
+
+/**
+ * The {@code PlainObjectBulkheadAspectExt} class provides an aspect for managing method executions
+ * that return plain objects while enforcing bulkhead constraints.
+ */
+public class PlainObjectBulkheadAspectExt implements BulkheadAspectExt {
+
+    private static final Logger logger = LoggerFactory.getLogger(PlainObjectBulkheadAspectExt.class);
+    private final TimeLimiterRegistry timeLimiterRegistry;
+    private final ScheduledExecutorService executorService;
+
+    public PlainObjectBulkheadAspectExt(TimeLimiterRegistry timeLimiterRegistry) {
+        this.timeLimiterRegistry = timeLimiterRegistry;
+        this.executorService = Executors.newScheduledThreadPool(Runtime.getRuntime().availableProcessors());
+    }
+
+    /**
+     * Determines if the aspect can handle the specified return type of the method.
+     *
+     * @param returnType the AOP method return type class
+     * @return {@code true} if the method has a plain object return type; {@code false} otherwise.
+     */
+    @Override
+    @SuppressWarnings("rawtypes")
+    public boolean canHandleReturnType(Class returnType) {
+        return !returnType.equals(CompletableFuture.class)
+                && !returnType.equals(Mono.class)
+                && !returnType.equals(Flux.class);
+    }
+
+    /**
+     * Handle the execution of a method wrapped in a bulkhead context.
+     *
+     * @param proceedingJoinPoint Spring AOP proceedingJoinPoint
+     * @param bulkhead the configured bulkhead
+     * @param methodName the method name
+     * @return the result object from the method execution
+     * @throws Throwable if an exception occurs during the method execution
+     */
+    @Override
+    public Object handle(ProceedingJoinPoint proceedingJoinPoint, Bulkhead bulkhead, String methodName) throws Throwable {
+        if (bulkhead instanceof ThreadPoolBulkheadAdapter) {
+            ThreadPoolBulkhead threadPoolBulkhead = ((ThreadPoolBulkheadAdapter) bulkhead).threadPoolBulkhead();
+            return handleThreadPoolBulkhead(proceedingJoinPoint, threadPoolBulkhead, methodName);
+        }
+
+        return handleSemaphoreBulkhead(proceedingJoinPoint, bulkhead, methodName);
+    }
+
+    private Object handleSemaphoreBulkhead(ProceedingJoinPoint proceedingJoinPoint, Bulkhead bulkhead, String methodName) {
+        TimeLimiter timeLimiter = timeLimiterRegistry.timeLimiter(bulkhead.getName());
+        try {
+            Supplier<CompletableFuture<Object>> futureSupplier = createBulkheadFuture(proceedingJoinPoint, bulkhead);
```

Comment:
You cannot time out a SemphoreBulkhead, since it is not running on a separate thread.
The SemphoreBulkhead already returns a plain object type.  It is handled here: https://github.com/resilience4j/resilience4j/blob/ce6cf63905099b57daec391a6b0a0d6dabbbbcf5/resilience4j-spring/src/main/java/io/github/resilience4j/bulkhead/configure/BulkheadAspect.java#L194

If we want to time out a call which is waiting for a permission, we would need to work with Semaphore timeouts.